internal/convert.go

```diff
@@ -343,6 +348,20 @@ func (conv *Conv) SampleBadRows(n int) []string {
 	return l
 }
 
+func (conv *Conv) AddShardIdColumn() {
+	for t, ct := range conv.SpSchema {
+		colName := shardIdColumn
+		columnId := GenerateColumnId()
+		ct.ColIds = append(ct.ColIds, columnId)
+		ct.ColDefs[columnId] = ddl.ColumnDef{Name: colName, Id: columnId, T: ddl.Type{Name: ddl.String, Len: 50}, NotNull: false}
+		ct.ShardIdColumn = columnId
+		conv.SpSchema[t] = ct
+		var issues []SchemaIssue
+		issues = append(issues, ShardIdColumnAdded, ShardIdColumnPrimaryKey)
+		conv.SchemaIssues[ct.Id].ColumnLevelIssues[columnId] = issues
+	}
+}
+
```

Comment:
Makes sense but IMO it would require a little more thought on the API design so will take it up when later when we are adding the feature to select an existing column as migration shard id column